stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+
+    Parameters
+    ----------
+    mean_type: string
+        The type of mean to be used for the Gaussian Process model
+        The parameters in log scale have a prefix of "log_"
+
+    Returns
+    -------
+        A list of the parameters for the mean for the GP model
+    """
+    if (mean_type == "gaussian") or (mean_type == "exponential"):
+        return ["log_A", "t0", "log_sig"]
+    elif mean_type == "constant":
+        return ["log_A"]
+    elif (mean_type == "skew_gaussian") or (mean_type == "skew_exponential"):
+        return ["log_A", "t0", "log_sig1", "log_sig2"]
+    elif mean_type == "fred":
+        return ["log_A", "t0", "delta", "phi"]
+    else:
+        raise ValueError("Mean type not implemented")
+
+
+def get_gp_params(kernel_type, mean_type):
+    """
+    Generates a list of the parameters for the GP model based on the kernel and mean type.
+    To be used to set the order of the parameters for `get_prior` and `get_likelihood` functions.
+
+    Parameters
+    ----------
+    kernel_type: string
+        The type of kernel to be used for the Gaussian Process model
+
+    mean_type: string
+        The type of mean to be used for the Gaussian Process model
+
+    Returns
+    -------
+        A list of the parameters for the GP model
+
+    Examples
+    --------
+    get_gp_params("QPO_plus_RN", "gaussian")
+    ['log_arn', 'log_crn', 'log_aqpo', 'log_cqpo', 'log_freq', 'log_A', 't0', 'log_sig']
+    """
+    kernel_params = _get_kernel_params(kernel_type)
+    mean_params = _get_mean_params(mean_type)
+    kernel_params.extend(mean_params)
+    return kernel_params
+
+
+def get_prior(params_list, prior_dict):
+    """
+    A prior generator function based on given values.
+    Makes a jaxns specific prior function based on the given prior dictionary.
+    Jaxns requires the parameters of the prior function and log_likelihood function to
+    be in the same order. This order is made according to the params_list.
+
+    Parameters
+    ----------
+    params_list:
+        A list in order of the parameters to be used.
+
+    prior_dict:
+        A dictionary of the priors of parameters to be used.
+        These parameters should be from tensorflow_probability distributions / Priors from jaxns
+        or special priors from jaxns.
+        **Note**: If jaxns priors are used, then the name given to them should be the same as
+        the corresponding name in the params_list.
+        Also, if a parameter is to be used in the log scale, it should have a prefix of "log_"
+
+    Returns
+    -------
+    The Prior generator function.
+    The arguments of the prior function are in the order of
+        Kernel arguments (RN arguments, QPO arguments),
+        Mean arguments
+        Miscellaneous arguments
+
+    Examples
+    --------
+    A prior function for a Red Noise kernel and a Gaussian mean function
+    Obain the parameters list
+    if not can_sample:
```

Comment:
I think it might be useful to add a comment here that these four lines are not part of the example, and only here to help execute the test suite

Reply:
This was just supposed to be an Example, not a doctest (Those lines were unnecessary). Have made some changes now.